packages/extras/src/lib/components/Gizmo/Gizmo.svelte

```diff
@@ -1,467 +1,143 @@
-    paddingX
-    paddingY
-    invalidate()
-  })
-
-  const orthoCam = new OrthographicCamera(-1.25, 1.25, 1.25, -1.25, 0, 4)
-  orthoCam.position.set(0, 0, 2)
+  const { camera, renderer, autoRenderTask, shouldRender, size, invalidate } = useThrelte()
 
-  const root = new Scene()
+  ref = new ViewportGizmo(camera.current as PerspectiveCamera | OrthographicCamera, renderer)
 
   const viewport = new Vector4()
 
+  const _controls = $derived($parent instanceof OrbitControls ? $parent : controls)
+
   useTask(
     renderTask?.key ?? Symbol('threlte-extras-gizmo-render'),
     () => {
-      const { autoClear, toneMapping } = renderer
-      renderer.autoClear = false
-      renderer.getViewport(viewport)
-      renderer.toneMapping = toneMapped ? renderer.toneMapping : 0
+      if (shouldRender()) {
+        const toneMapping = renderer.toneMapping
+
+        renderer.getViewport(viewport)
+        renderer.toneMapping = NoToneMapping
 
-      const x =
-        horizontalPlacement === 'left'
-          ? paddingX
-          : renderer.domElement.offsetWidth - size - paddingX
-      const y =
-        verticalPlacement === 'bottom'
-          ? paddingY
-          : renderer.domElement.offsetHeight - size - paddingY
+        ref.render()
 
-      renderer.setViewport(x, y, size, size)
-      renderer.render(root, orthoCam)
-      renderer.setViewport(viewport.x, viewport.y, viewport.z, viewport.w)
-      renderer.autoClear = autoClear
-      renderer.toneMapping = toneMapping
+        renderer.setViewport(viewport)
+        renderer.toneMapping = toneMapping
+      }
     },
     {
-      ...(renderTask ?? { after: autoRenderTask }),
-      autoInvalidate: false
+      autoInvalidate: false,
+      ...(renderTask ?? { after: autoRenderTask })
     }
   )
 
-  // User interaction must be handled manually because
-  // the gizmo is not in the main scene. The click
-  // target is added as a sibling of the renderer's
-  // dom element.
-  const clickTarget = document.createElement('div')
-  // We need to know the bounding rect of the renderer's dom element
-  const renderTarget = renderer.domElement
-  const boundingRect = renderTarget.getBoundingClientRect()
-
-  clickTarget.style.position = 'absolute'
-
   $effect.pre(() => {
-    if (horizontalPlacement === 'right') {
-      clickTarget.style.right = ''
-      clickTarget.style.left = `${boundingRect.right - size - paddingX}px`
-    } else {
-      clickTarget.style.right = ''
-      clickTarget.style.left = `${paddingX + boundingRect.left}px`
-    }
-
-    if (verticalPlacement === 'bottom') {
-      clickTarget.style.bottom = ''
-      clickTarget.style.top = `${boundingRect.bottom - size - paddingY}px`
-    } else {
-      clickTarget.style.bottom = ''
-      clickTarget.style.top = `${paddingY + boundingRect.top}px`
-    }
-
-    clickTarget.style.height = `${size}px`
-    clickTarget.style.width = `${size}px`
+    ref.camera = $camera as PerspectiveCamera | OrthographicCamera
```

Comment:
so what i've been doing in some components is allow an optional camera prop that if not present will default to the camera in `useThrelte` something like this

```svelte
<script lang="ts">
  type Props = { camera: PerspectiveCamera | OrthographicCamera }
  let { camera, ref } = props()
  
  const { camera: defaultCamera } = useThrelte()
  
  // can't remember if defaultCamera is a store or has a `current` prop
  const _camera = $derived(camera ?? $defaultCamera)
  ref = new ViewportGizmo(camera.current)
  $effect(() => {
    if (_camera !== ref.camera) {
      ref.camera = _camera
    }
  })
</script>
```

should we do the same here? i don't think you'd ever want to control a camera that wasn't the one used for rendering but maybe???

Reply:
camera arg added!